src/main/java/com/team1806/frc2020/subsystems/Hood.java

```diff
@@ -0,0 +1,182 @@
+
+
+public class Hood extends Subsystem {
+
+    enum HoodControlState {
+
+        kIdle, kPositionControl, kHoldPosition, kManualControl
+    }
+
+
+    private class PeriodicIO {
+        public double timestamp;
+        public double motorVoltage;
+        public double motorAmps;
+        public double currentPosition;
+        public double currentSpeed;
+        public double wantedPosition;
+        public double wantedManualMovement;
+        public HoodControlState HoodState;
+
+    }
+
+
+
+    private static Hood HOOD = new Hood();
+
+    private HoodControlState mHoodControlState;
+    private TalonSRX mCANTalonSRX;
+    private PeriodicIO mPeriodicIO;
+    private ReflectingCSVWriter mCSVWriter;
+
+
+    private Hood(){
+        mCANTalonSRX = new TalonSRX(Constants.kHoodMotor);
+        mPeriodicIO = new PeriodicIO();
+        mHoodControlState = HoodControlState.kIdle;
+        mCANTalonSRX.setNeutralMode(NeutralMode.Brake);
+    }
+
+    public static Hood GetInstance(){
+        return HOOD;
+    }
+
+    private double ConvertEncoderClicksToAngle(double encoderCount){
+        return encoderCount * Constants.kHoodDegreesPerCount;
+    }
+
+    private double ConvertAngleToEncoderClicks(double angle){
+        return angle / Constants.kHoodDegreesPerCount;
+    }
+
+    public void setWantedAngle(double angle){
+        setControlState(HoodControlState.kPositionControl);
+        mPeriodicIO.wantedPosition = angle;
+    }
+
+    public void setManualControl(double wantedPosition){
+        setControlState(HoodControlState.kManualControl);
+        mPeriodicIO.wantedManualMovement = wantedPosition;
+    }
+
+    public void setWantIdle(){
+        setControlState(HoodControlState.kIdle);
+        mPeriodicIO.wantedPosition = 0.0;
+        mPeriodicIO.wantedManualMovement = 0.0;
+    }
+
+    public boolean isOnTarget() {
+        return Math.abs(mPeriodicIO.wantedPosition - mPeriodicIO.currentPosition) < Constants.kHoodAcceptableAngleDeviation && Math.abs(mPeriodicIO.currentSpeed) <= Constants.kHoodAcceptableSpeed;
+    }
+
+    private void reloadGains(){
+        if(mCANTalonSRX!= null){
+            mCANTalonSRX.config_kP(0, Constants.kHoodPositionControlKp);
+            mCANTalonSRX.config_kI(0, Constants.kHoodPositionControlKi);
+            mCANTalonSRX.config_kD(0,Constants.kHoodPositionControlKd);
+            mCANTalonSRX.config_kF(0, Constants.kHoodPositionControlKf);
+        }
+    }
+
+    public void writePeriodicOutputs(){
+        switch (mPeriodicIO.HoodState){
+            default:
+            case kIdle:
+                mCANTalonSRX.set(ControlMode.PercentOutput, 0);
+                break;
+            case kPositionControl:
+                if(isOnTarget()){
+                    setControlState(HoodControlState.kHoldPosition);
+                }
+                else{
+                    mCANTalonSRX.set(ControlMode.Position, mPeriodicIO.wantedPosition);

```

Comment:
wanted position is an angle, you're setting encoder clicks. Check this on Turret too. Also maybe rename wantedPosition to wantedAngle in both.